ERC20-warmup/fathalian/IC3Token.sol

```diff
@@ -0,0 +1,95 @@
+pragma solidity ^0.4.10;
+
+
+contract IC3Token {
+
+    uint8 public constant decimals = 18;
+
+    string public constant name = "IC3 2017 Bootcamp Token";
+
+    string public constant symbol = "IC3";
+
+    mapping (address => uint256) public balances;
+
+    mapping (address => mapping (address => uint256)) public approvals;
+
+    uint256 public totalSupply;
+
+    event Transfer(address indexed _from, address indexed _to, uint256 _value);
+
+    event Approval(address indexed _owner, address indexed _spender, uint256 _value);
+
+    function balanceOf(address _owner) constant returns (uint256 balance) {
+        return balances[_owner];
+    }
+
+
+    function approve(address _spender, uint256 _value) returns (bool success){
+        if (msg.sender == _spender) return false;
+
+        approvals[msg.sender][_spender] = _value;
+
+        Approval(msg.sender, _spender, _value);
+        return true;
+    }
+
+
+    function allowance(address _owner, address _spender) constant returns (uint256 remaining) {
+        remaining = approvals[_owner][_spender];
+        return remaining;
+    }
+
+    function transfer(address _to, uint256 _value) returns (bool success) {
+        if (balances[msg.sender] < _value) return false;
```

Comment:
Good to check, but safeSubtract will also check this so can be safely removed :).